ext/pdo_firebird/firebird_statement.c

```diff
@@ -372,7 +372,9 @@ static int firebird_stmt_get_col(pdo_stmt_t *stmt, int colno, char **ptr,  /* {{
 
 			*ptr = FETCH_BUF(S->fetch_buf[colno], char, CHAR_BUF_LEN, NULL);
 
-			if (n >= 0) {
+			if((var->sqltype & ~1)==SQL_DOUBLE) {
```

Comment:
Nit: there should be spaces after `if` and around the `==` operator.

Reply:
done